internal/controller/pod_controller.go

```diff
@@ -81,51 +84,68 @@ func (a *PodAdmissionWebhook) Handle(ctx context.Context, req admission.Request)
 // handleCreatePodRequest Finds relevant AuthProxyWorkload resources and updates the pod
 // with matching resources, returning a non-nil pod when the pod was updated.
 func (a *PodAdmissionWebhook) handleCreatePodRequest(ctx context.Context, p corev1.Pod) (*corev1.Pod, error) {
+	l := logf.FromContext(ctx)
```

Comment:
This change moves code from the `handleCreatePodRequest()` function into a new function called `findMatchingProxies()` so that it can be reused between the webhook and the new Pod event listener.